dist/aframe-altspace-component.js

```diff
@@ -465,86 +467,744 @@
+	{
+		schema: {
+			author: { type: 'string', default: null },
+			app: { type: 'string', default: null },
+			instance: { type: 'string', default: null },
+			'ref-url': { type: 'string', default: null },
+		},
+		init: function() {
+			var component = this;
+
+			altspace.utilities.sync.connect({
+				authorId: this.data.author,
+				appId: this.data.app,
+				instanceId: this.data.instance,
+				baseRefUrl: this.data['ref-url']
+			}).then(function(connection) {
+				this.connection = connection;
+				this.sceneEl.emit('connected', null, false);
+
+				this.sceneRef = this.connection.instance.child('scene');
+				this.clientsRef = this.connection.instance.child('clients');
+
+				// temporary way of having unique identifiers for each client
+				this.clientId = this.sceneEl.object3D.uuid;
+				var masterClientId;
+				this.clientsRef.on("value", function (snapshot) {
+					var clientIds = snapshot.val();
+
+					var masterClientKey = Object.keys(clientIds)[0];
+					masterClientId = clientIds[masterClientKey];
+				});
+
+				this.clientsRef.on('child_added', function(childSnapshot) {
+					var joinedClientId = childSnapshot.val();
+					//let the master client flag get set first
+					setTimeout(function(){
+						component.sceneEl.emit('clientjoined', {id: joinedClientId}, false);
+					}, 1);
+				});
+
+				this.clientsRef.on('child_removed', function(childSnapshot) {
+					var leftClientId = childSnapshot.val();
+					//let the master client flag get set first
+					setTimeout(function(){
+						component.sceneEl.emit('clientleft', {id: leftClientId}, false);
+					}, 1);
+				});
+
+				// add our client ID to the list of connected clients, 
+				// but have it be automatically removed by firebase if we disconnect for any reason
+				this.clientsRef.push(this.clientId).onDisconnect().remove();
+
+				this.connection.instance.child('initialized').once('value', function (snapshot) {
+					var shouldInitialize = !snapshot.val();
+					snapshot.ref().set(true);
+
+					this.sceneEl.emit('instanceready', { shouldInitialize: shouldInitialize }, false);
+				}.bind(this));
+
+
+				Object.defineProperty(this, 'isMasterClient', {
+					get: function () { return masterClientId === this.clientId; }.bind(this)
+				});
+			}.bind(this));
+		}
+	});
+
+	AFRAME.registerComponent('sync',
+	{
+		schema: {
+			mode: { default: 'link' }
+		},
+		init: function () {
+			var scene = document.querySelector('a-scene');
+			var syncSys = scene.systems['sync-system'];
+
+			var ref;
+			var key;
+			var dataRef;
+			var ownerRef;
+			var ownerId;
+			var isMine = false;
+
+			var component = this;
+
+			//Make sure someone always owns an object. If the owner leaves and we are the master client, we will take it.
+			//This ensures, for example, that synced animations keep playing
+			scene.addEventListener('clientleft', function(event){
+				var shouldTakeOwnership = (!ownerId || ownerId === event.detail.id) && syncSys.isMasterClient;
```

Comment:
That !ownerId part isn't necessary and I removed it. OwnerId is the current owner. So it's saying "if the owner left, and I'm the master client, then take ownership"